include/grpcpp/impl/codegen/call.h

```diff
@@ -598,12 +792,353 @@ class CallOpSetInterface : public CompletionQueueTag {
+  }
+
+  virtual void AddInterceptionHookPoint(
+      experimental::InterceptionHookPoints type) override {
+    hooks_[static_cast<int>(type)] = true;
+  }
+
+  virtual ByteBuffer* GetSendMessage() override { return send_message_; }
+
+  virtual std::multimap<grpc::string, grpc::string>* GetSendInitialMetadata()
+      override {
+    return send_initial_metadata_;
+  }
+
+  virtual Status GetSendStatus() override {
+    return Status(static_cast<StatusCode>(*code_), *error_message_,
+                  *error_details_);
+  }
+
+  virtual void ModifySendStatus(const Status& status) override {
+    *code_ = static_cast<grpc_status_code>(status.error_code());
+    *error_details_ = status.error_details();
+    *error_message_ = status.error_message();
+  }
+
+  virtual std::multimap<grpc::string, grpc::string>* GetSendTrailingMetadata()
+      override {
+    return send_trailing_metadata_;
+  }
+
+  virtual void* GetRecvMessage() override { return recv_message_; }
+
+  virtual std::multimap<grpc::string_ref, grpc::string_ref>*
+  GetRecvInitialMetadata() override {
+    return recv_initial_metadata_->map();
+  }
+
+  virtual Status* GetRecvStatus() override { return recv_status_; }
+
+  virtual std::multimap<grpc::string_ref, grpc::string_ref>*
+  GetRecvTrailingMetadata() override {
+    return recv_trailing_metadata_->map();
+  }
+
+  virtual void SetSendMessage(ByteBuffer* buf) override { send_message_ = buf; }
+
+  virtual void SetSendInitialMetadata(
+      std::multimap<grpc::string, grpc::string>* metadata) override {
+    send_initial_metadata_ = metadata;
+  }
+
+  virtual void SetSendStatus(grpc_status_code* code,
+                             grpc::string* error_details,
+                             grpc::string* error_message) override {
+    code_ = code;
+    error_details_ = error_details;
+    error_message_ = error_message;
+  }
+
+  virtual void SetSendTrailingMetadata(
+      std::multimap<grpc::string, grpc::string>* metadata) override {
+    send_trailing_metadata_ = metadata;
+  }
+
+  virtual void SetRecvMessage(void* message) override {
+    recv_message_ = message;
+  }
+
+  virtual void SetRecvInitialMetadata(internal::MetadataMap* map) override {
+    recv_initial_metadata_ = map;
+  }
+
+  virtual void SetRecvStatus(Status* status) override { recv_status_ = status; }
+
+  virtual void SetRecvTrailingMetadata(internal::MetadataMap* map) override {
+    recv_trailing_metadata_ = map;
+  }
+
+  virtual std::unique_ptr<ChannelInterface> GetInterceptedChannel() override {
+    auto* info = call_->client_rpc_info();
+    if (info == nullptr) {
+      return std::unique_ptr<ChannelInterface>(nullptr);
+    }
+    // The intercepted channel starts from the interceptor just after the
+    // current interceptor
+    return std::unique_ptr<ChannelInterface>(new internal::InterceptedChannel(
+        reinterpret_cast<grpc::ChannelInterface*>(info->channel()),
+        curr_iteration_ + 1));
```

Comment:
Maybe call it `current_interceptor_index_` rather than `curr_iteration_`

Reply:
done! (not yet pushed)